ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+     * @param country the string corresponding to the country where the iidm component is
+     */
+    void convertIidmIdToUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode = null;
+        String voltageNameOrId = voltageLevel.getName() != null ? voltageLevel.getName() : voltageLevel.getId();
+        int busbar = 'a';
+        do {
+            if (busbar <= 'z') {
+                ucteNodeCode = new UcteNodeCode(
+                        UcteCountryCode.valueOf(country),
+                        voltageNameOrId,
+                        voltageLevelCodeFromIidmVoltage(voltageLevel.getNominalV()),
+                        (char) busbar
+                );
+            } else {
+                LOGGER.warn("There is more than 26 nodes with the the id starting by '{}'", ucteNodeCode);
+                generateUcteNodeCode(id, voltageLevel, country);
+                return;
+            }
+            busbar++;
+        } while (iidmIdToUcteNodeCodeId.values().contains(ucteNodeCode));
+        iidmIdToUcteNodeCodeId.put(id, ucteNodeCode);
+    }
+
+    /**
+     * create a unique UcteNodeCode using the voltage level and the country given and
+     * store it to {@link UcteExporter#iidmIdToUcteNodeCodeId} so we don't need to create it every time
+     * and ensure  its uniqueness.
+     * UcteNodeCode id = 1 character representing the country + 2 characters for the substation (or "aa" if the substation
+     * doesn't have a name) + 3 character generated from "aaa" to "zzz" + 1 character for the voltage level code and a character
+     * generated from "a" to "z". The goal is to create a ucte compliant id from any iidm component with non ucte compliant ids
+     *
+     * @param id the orignal id of the iidm component we want to convert
+     * @param voltageLevel the {@link VoltageLevel} of the component we want to convert
+     * @param country the string corresponding to the country where the iidm component is
+     */
+    void generateUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode = null;
+        String substationName = null;
+
+        if (voltageLevel.getSubstation().getName() != null && voltageLevel.getSubstation().getName().length() >= 2) {
+            substationName = voltageLevel.getSubstation().getName().substring(0, 2);
+        } else {
+            substationName = "aa";
+        }
+        do {
+            generatedGeographicalName = incrementGeneratedGeographicalName(generatedGeographicalName);
+            ucteNodeCode = new UcteNodeCode(
+                    UcteCountryCode.valueOf(country),
+                    substationName + generatedGeographicalName.substring(0, 3),
+                    voltageLevelCodeFromIidmVoltage(voltageLevel.getNominalV()),
+                    generatedGeographicalName.charAt(3)
+            );
+        } while (iidmIdToUcteNodeCodeId.values().contains(ucteNodeCode));
```

Comment:
I'm not sure this is really efficient. Maybe you need an UCTE dictionary of the existing nodes IDs